### PR TITLE
Support JWT(no signature)

### PR DESCRIPTION
**Pull request by chenyukaola**

### CODE REVIEW

File: src/jwt/jwtparser.c [hunk 1]
```diff
@@ -154,21 +217,47 @@ JWS *JWTParser_Parse(const char *token)
...
+        return NULL;
+    }
 
+    // find the indexes of the dots
+    for (int i = 0; i < strlen(token) && idx < 2; ++i) {
+        if (token[i] == '.')
+            dots[idx++] = i;
+    }
 
+    if (0 == dots[0] || strlen(token) == dots[0] || 0 == dots[1]) {
```
Comment (stiartsly) [inline src/jwt/jwtparser.c:254]: check "idex == 3" (3段内容 header.payload.signature  或者 header.payload.

Comment (stiartsly) [reply src/jwt/jwtparser.c:254]: hard to understand the condition : "(0 == dots[0] || strlen(token) == dots[0] || 0 == dots[1]) "
I'd rather to check !(dots[0] > 0 &&  dots[1] > dots[0] && dots[1] < strlen(token) )

File: src/jwt/jwtparser.c [hunk 2]
```diff
@@ -59,13 +59,19 @@ static cjose_jwk_t *get_jwk(JWS *jws)
...
-    keyid = DIDURL_FromString(JWS_GetKeyId(jws), NULL);
-    if (!keyid)
-        goto errorExit;
-
-    key = DIDDocument_GetPublicKey(doc, keyid);
-    if (!key)
-        goto errorExit;
+    if (!JWS_GetKeyId(jws)) {
+        keyid = DIDDocument_GetDefaultPublicKey(doc);
+        key = DIDDocument_GetPublicKey(doc, keyid);
```
Comment (stiartsly) [inline src/jwt/jwtparser.c:64]: Same with "else" branch to destroy "keyid" ?

File: src/jwt/jwtparser.c [hunk 3]
```diff
@@ -59,13 +59,19 @@ static cjose_jwk_t *get_jwk(JWS *jws)
...
+        if (!key)
+            goto errorExit;
+    }
+    else {
+        keyid = DIDURL_FromString(JWS_GetKeyId(jws), issuer);
+        key = DIDDocument_GetPublicKey(doc, keyid);
+        DIDURL_Destroy(keyid);
+        if (!key)
+            goto errorExit;
+    }
```
Comment (stiartsly) [inline src/jwt/jwtparser.c:74]: Or we write these ones:
  if (!JWS_GetKeyId(jws)) {
      ...
  } else {
      ...
  }
  DIDURL_Destroy(keyid);
  if (!key) 
         goto errorExit;

File: src/jwt/jwtparser.c (hunk 2 above)
Comment (chenyukaola) [reply src/jwt/jwtparser.c:64]:
```c
DIDDocument_GetDefaultPublicKey
```
The return value from this api is the point from 'doc', not create the new one, so no need to free the return value, which is different with 'DIDURL_NewFromDid'.

Comment (chenyukaola) [reply src/jwt/jwtparser.c:64]: > Same with "else" branch to destroy "keyid" ?

The return value from this api is the point from 'doc', not create the new one, so no need to free the return value, which is different with 'DIDURL_NewFromDid'.

File: src/jwt/jwtparser.c (hunk 3 above)
Comment (chenyukaola) [reply src/jwt/jwtparser.c:74]: > Or we write these ones:
> if (!JWS_GetKeyId(jws)) {
> ...
> } else {
> ...
> }
> DIDURL_Destroy(keyid);
> if (!key)
> goto errorExit;

no,  one branch is not destroy 'keyed'.

**Pull request merged**